hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/AvroKafkaSource.java

```diff
@@ -23,6 +23,7 @@
 import org.apache.hudi.utilities.schema.SchemaProvider;
 import org.apache.hudi.utilities.sources.helpers.KafkaOffsetGen;
 import org.apache.hudi.utilities.sources.helpers.KafkaOffsetGen.CheckpointUtils;
+import org.apache.hudi.utilities.sources.serde.HoodieAvroKafkaDeserializer;
 
 import io.confluent.kafka.serializers.KafkaAvroDeserializer;
```

Comment:
@pratyakshsharma I thought this is what this deserializer does. no?

Reply:
No, it reads the encoded schema id from the byte buffer and simply uses that schema for reading. It does not take into account readerSchema.